staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -483,11 +468,13 @@ See http://k8s.io/docs/reference/using-api/api-concepts/#conflicts`, err)
 
 			if !o.DryRun {
 				// Then create the resource and skip the three-way merge
-				options := metav1.CreateOptions{}
+				helper := resource.NewHelper(info.Client, info.Mapping)
 				if o.ServerDryRun {
-					options.DryRun = []string{metav1.DryRunAll}
+					if err := resource.VerifyDryRun(info.Mapping.GroupVersionKind, o.DynamicClient, o.DiscoveryClient); err != nil {
```

Comment:
Same here, we can't ignore the error, otherwise it's going to apply the object :-/

Reply:
Fixed!